lib/utils/compose_ts.ts

```diff
@@ -1405,6 +1418,18 @@ export function createRunLoop(tick: (...args: any[]) => void) {
 	return runloop;
 }
 
+async function tryGetContractContent(
+	filePath: string,
+): Promise<any | undefined> {
+	try {
+		const fileContentAsString = await fs.readFile(filePath, 'utf8');
+		return jsyaml.load(fileContentAsString) as any;
```

Comment:
Does the builder have any logic to validate contracts, like a schema? If not, at least a basic sanity test to check that this is not a blank string or null / undefined or empty object, and also a test that differentiates between:
* `./balena.yml` and
* `./.balena/balena.yml` - this one representing build secrets and variables and documented at https://www.balena.io/docs/learn/deploy/deployment/#build-time-only-secret-file

Users are likely to swap these files at some point, so it would be good to have at least some minimal validation. We have already discussed the intention of unifying these two `balena.yml` (while somehow preserving backwards compatibility) in two product calls, but the work hasn't been done. You're super welcome to do it if you feel like! Some more details on this brainstorm topic: https://jel.ly.fish/90f5a591-db7e-4a1a-8f72-1a38f203e3f2

Reply:
Looks like the only required field in a contract is `type`, with a  descriptive error message this might be enough to identify and clue the user about the wrong configuration path

https://www.balena.io/docs/learn/deploy/deploy-with-balena-button/#balenayml-configuration-file
https://github.com/product-os/contrato/blob/master/lib/types/definitions.cue